gems/aws-sdk-core/lib/aws-sdk-core/plugins/protocols/api_gateway.rb

```diff
@@ -4,9 +4,24 @@ module Aws
   module Plugins
     module Protocols
       class ApiGateway < Seahorse::Client::Plugin
+
+        class ContentTypeHandler < Seahorse::Client::Handler
+          def call(context)
+            body = context.http_request.body
+            # Rest::Handler will set a default JSON body, so size can be checked
+            # if this handler is run after serialization.
+            if body.respond_to?(:size) && body.size > 0
```

Comment:
Needs the same logic as in the http handler - if body doesn't respond to size, we need to assume there may be streaming content.